source/class/qx/tool/cli/Utils.js

```diff
@@ -0,0 +1,105 @@
+/* ************************************************************************
+
+   qooxdoo - the new era of web development
+
+   http://qooxdoo.org
+
+   Copyright:
+     2017-202 The authors
+
+   License:
+     MIT: https://opensource.org/licenses/MIT
+     See the LICENSE file in the project's top-level directory for details.
+
+   Authors:
+     * John Spackman (john.spackman@zenesis.com, @johnspackman)
+     * Christian Boulanger (info@bibliograph.org, @cboulanger)
+
+************************************************************************ */
+
+const child_process = require("child_process");
+
+qx.Class.define("qx.tool.cli.Utils", {
+  type: "static",
+  members: {
+    /**
+     * Awaitable wrapper around child_process.spawn.
+     * Runs a command in a separate process. The output of the command
+     * is ignored. Throws when the exit code is not 0.
+     * @param  {String} cmd Name of the command
+     * @param  {Array} args Array of arguments to the command
+     * @return {Promise<Number>} A promise that resolves with the exit code
+     */
+    run(cmd, args) {
+      let opts = {env: process.env};
+      return new Promise((resolve, reject) => {
+        let exe = child_process.spawn(cmd, args, opts);
+        // suppress all output unless in verbose mode
+        exe.stdout.on("data", data => {
+          if (this.argv.verbose) {
+            qx.tool.compiler.Console.log(data.toString());
+          }
+        });
+        exe.stderr.on("data", data => {
+          if (this.argv.verbose) {
+            qx.tool.compiler.Console.error(data.toString());
+          }
+        });
+        exe.on("close", code => {
+          if (code !== 0) {
+            let message = `Error executing '${cmd} ${args.join(" ")}'. Use --verbose to see what went wrong.`;
+            throw new qx.tool.utils.Utils.UserError(message);
```

Comment:
Should it throw here, or better to call `reject`?

Reply:
done.